### PR TITLE
Refine variant reason computation to include SPLIT and STATIC

### PR DESCRIPTION
**Pull request by typotter**
# What Does This Do

Computes the correct OpenFeature evaluation reason from the UFC allocation structure instead of hardcoding `TARGETING_MATCH` for all successful evaluations.

Three-way logic in `resolveVariant()`:
- `STATIC` — no rules, no shards (unconditional allocation)
- `SPLIT` — no rules, shards present (percentage-based traffic split)
- `TARGETING_MATCH` — rules present (flag matched a targeting rule)

Also fixes one test assertion that needed updating after rebasing onto #10990: a null targeting key on a static flag now correctly evaluates as `STATIC` rather than `TARGETING_MATCH`.

# Motivation

System tests `Test_FFE_Eval_Metric_Basic` (expects `reason=static`) and `Test_FFE_Eval_Reason_Split` (expects `reason=split`) were failing because the evaluator hardcoded `TARGETING_MATCH` regardless of allocation structure. This aligns Java behavior with the Python and Go SDK reference implementations.

Co-authored-by: Leo Romanovsky <leo.romanovsky@datadoghq.com> (original fix in #10971)

# Additional Notes

The `split` object is passed into `resolveVariant()` so the reason can be derived at resolution time without additional lookups.

The semantic question of whether a catch-all (no-rules, no-shards) allocation inside a multi-allocation flag should return `STATIC` vs another reason is out of scope here — that's a pre-existing behavior question tracked separately.

# Contributor Checklist

- Format the title according to [the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any other useful labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- Avoid using `close`, `fix`, or [any linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- Update the [CODEOWNERS](https://github.com/DataDog/dd-trace-java/blob/master/.github/CODEOWNERS) file on source file addition, migration, or deletion
- Update [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) with any new configuration flags or behaviors

### CODE REVIEW

File: products/feature-flagging/feature-flagging-api/src/main/java/datadog/trace/api/openfeature/DDEvaluator.java [hunk 1]
```diff
@@ -351,7 +359,10 @@ private static <T> ProviderEvaluation<T> resolveVariant(
     final ProviderEvaluation<T> result =
         ProviderEvaluation.<T>builder()
             .value(mapValue(target, variant.value))
-            .reason(Reason.TARGETING_MATCH.name())
+            .reason(
+                !isEmpty(allocation.rules)
+                    ? Reason.TARGETING_MATCH.name()
+                    : !isEmpty(split.shards) ? Reason.SPLIT.name() : Reason.STATIC.name())
```
Comment (sameerank) [inline products/feature-flagging/feature-flagging-api/src/main/java/datadog/trace/api/openfeature/DDEvaluator.java:391]: 👍

**Review by sameerank — APPROVED** [review]
